space-concordia-robotics/robotics-prototype
# [#256] Add two ros action server/client demos

# Assignee Section

## Description
Added: robotics-prototype/robot/demos/RosAction
This contains a ROS package with two ROS Action Client/Server demos:
One for calculating a set of numbers in the Fibonacci sequence
One for setting the number of iterations a for loop runs in order to 'wash dishes'

Note that there is additional info on building a ROS Action Server/Client pair located inside of the README_long.md file.

Added ROS Action Server/Client Demos
- Fibonacci Action
- Dishwasher Action

Test Feature 

1) cd into robotics-prototype/robot/demos/RosAction, run catkin_make
2) Run "source devel/setup.bash"
3) Run roscore
4) Open a new terminal

--For Fibonacci Action--
5) Run "rosrun actionlib_tutorials fibonacci_server.py"
6) Run "rosrun actionlib_tutorials fibonacci_client.py"

NOTE: Make sure you have run step 2 in the terminal you are using to echo these topics
To see the progress of the action while it runs, run "rostopic echo fibonacci/feedback"
To see the goal of the action, run "rostopic echo fibonacci/goal"
To see the result of the action, run "rostopic echo fibonacci/result"

Optional steps:
To change the amount of numbers in the Fibonacci Sequence to calculate, open RosAction/src/actionlib_tutorials/fibonacci_client.py and in line 22:
`goal = actionlib_tutorials.msg.FibonacciGoal(order=5)` change the value of order

To change the rate (in Hz) at which new numbers are calculated in the Fibonacci Sequence, open RosAction/src/actionlib_tutorials/fibonacci_server.py and in line 19:
`r = rospy.Rate(1)` change the value in the Rate() function

--For Dishwasher Action--
5) Run "rosrun dishwasher_action dishwasher_server.py"
6) Run "rosrun dishwasher_action dishwasher_client.py"

NOTE: Make sure you have run step 2 in the terminal you are using to echo these topics
To see the progress of the action while it runs, run "rostopic echo dishwasher/feedback"
To see the goal of the action, run "rostopic echo dishwasher/goal"
To see the result of the action, run "rostopic echo dishwasher/result"

Optional steps:
To change the amount of dishes to wash, open RosAction/src/dishwasher_action/dishwasher_client.py and in line 22:
`goal = dishwasher_action.msg.DishwasherGoal(dirtyDishes=50)` change the value of dirtyDishes

To change the rate (in Hz) at which dishes are washed, open RosAction/src/dishwasher_action/dishwasher_server.py and in line 19:
`r = rospy.Rate(2)` change the value in the Rate() function


closes #256 

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
